src/viadot/orchestration/prefect/utils.py

```diff
@@ -28,6 +28,394 @@
+        }.get(unit)
+
+        return full_date
+
+    def create_date_dict(self) -> Dict[str, str]:
+        """
+        Create and return a key phrase: dynamic date value dictionary.
+        Dictionary values "today", "yesterday" and "last year previous month" are formatted into the dynamic_date_format.
+        The other values and their formatting:
+            - "this month" - A string date formatted with a string format '%m'.
+            - "last month" - A string date formatted with a format "%mm"
+            - "last year" - A string date formatted with a string format '%Y'
+            - "now time" - A string date formatted with a string format '%H%M%S'.
+
+        Returns:
+            Dict[str, str]: A dictionary with key phrases as keys and dynamically created dates as values.
+        """
+        today = pendulum.today(self.dynamic_date_timezone)
+        yesterday = pendulum.yesterday(self.dynamic_date_timezone)
+        last_month = today.subtract(months=1).month
+        last_year = today.subtract(years=1)
+        now_time = pendulum.now(self.dynamic_date_timezone)
+        last_day_prev_month = today.subtract(months=1).end_of("month")
+        replacements = {
+            "today": today.strftime(self.dynamic_date_format),
+            "yesterday": yesterday.strftime(self.dynamic_date_format),
+            "this month": today.strftime("%m"),
+            "last month": f"{last_month:02d}",
+            "this year": today.strftime("%Y"),
+            "last year": last_year.strftime("%Y"),
+            "now time": now_time.strftime("%H%M%S"),
+            "last day previous month": last_day_prev_month.strftime(
+                self.dynamic_date_format
+            ),
+        }
+        return replacements
+
+    def handle_singular_dates(self, match: List[tuple], key: str) -> pendulum.DateTime:
+        """
+        Directs execution of a specific function based on the provided value of `key`.
+        Returns a pendulum.DateTime date generated based on `match`.
+
+        Args:
+            match (List[tuple]): List of every pattern match that occurs in a given string.
+            key (str): Key phrase that determines the execution of a specific function.
+
+        Returns:
+            pendulum.DateTime: A dynamically created date.
+        """
+        if key == "last_day_of_month":
+            for month_name, year in match:
+                replacement = self.process_last_day_of_month(year, month_name)
+
+        elif key == "x_units_ago_full_date":
+            for x, units in match:
+                x = int(x)
+                replacement = self.get_date_x_ago_full_date(int(x), units)
+
+        elif key == "x_years_ago":
+            for x in match:
+                x = int(x)
+                replacement = self.process_x_years_ago(x)
+        return replacement
+
+    def generate_dates_based_on_unit(self, number: int, unit: str) -> List[str]:
```

Comment:
mark this function internal by adding `_`.
```suggestion
    def _generate_dates_based_on_unit(self, number: int, unit: str) -> List[str]:
```